util/util.go

```diff
@@ -33,3 +37,20 @@ func GetCertPool(paths []string, system_roots bool) (*x509.CertPool, error) {
 	}
 	return pool, nil
 }
+
+// GetFilesMetadataHash returns base64-encoded hash of (size + name + modtime) of the file at path
+func GetFilesMetadataHash(paths []string) (string, error) {
+	hashData := ""
+
+	for _, path := range paths {
+		meta, err := os.Stat(path)
+		if err != nil {
+			return "", fmt.Errorf("couldn't stat '%s': %v", path, err)
+		}
+
+		hashData += meta.Name() + strconv.FormatInt(meta.Size(), 10) + meta.ModTime().String()
```

Comment:
probably makes sense to use `make([]string, 0, len(paths)` to store this, and then sort that slice before doing a `strings.Join` followed by hash.

Reply:
That makes sense, some calls to the function could use different order of paths, but there's no need to return different hash in that case. :+1: